modules/core/src/main/scala/lucuma/odb/api/model/ObservationModel.scala

```diff
@@ -140,44 +146,56 @@ object ObservationModel extends ObservationOptics {
   }
 
   final case class Edit(
-    observationId:       Observation.Id,
-    existence:           Input[Existence]                      = Input.ignore,
-    name:                Input[NonEmptyString]                 = Input.ignore,
-    status:              Input[ObsStatus]                      = Input.ignore,
-    activeStatus:        Input[ObsActiveStatus]                = Input.ignore,
-    asterismId:          Input[Asterism.Id]                    = Input.ignore,
-    targetId:            Input[Target.Id]                      = Input.ignore,
-    constraintSet:       Option[ConstraintSetModel.Edit]       = None,
-    scienceRequirements: Option[ScienceRequirementsModel.Edit] = None
+    observationId:        Observation.Id,
+    existence:            Input[Existence]                       = Input.ignore,
+    name:                 Input[NonEmptyString]                  = Input.ignore,
+    status:               Input[ObsStatus]                       = Input.ignore,
+    activeStatus:         Input[ObsActiveStatus]                 = Input.ignore,
+    asterismId:           Input[Asterism.Id]                     = Input.ignore,
+    targetId:             Input[Target.Id]                       = Input.ignore,
+    constraintSet:        Option[ConstraintSetModel.Edit]        = None,
+    scienceRequirements:  Option[ScienceRequirementsModel.Edit]  = None,
+    scienceConfiguration: Input[ScienceConfigurationModelEdit]   = Input.ignore
   ) {
 
-    def id: Observation.Id =
+    val id: Observation.Id =
       observationId
 
-    def pointing: ValidatedInput[(Input[Asterism.Id], Input[Target.Id])] =
+    val pointing: ValidatedInput[(Input[Asterism.Id], Input[Target.Id])] =
       (asterismId, targetId) match {
         case (Assign(_), Assign(_)) => InputError.fromMessage(s"Cannot assign both an asterism and a target to the observation").invalidNec
         case _                      => (asterismId, targetId).validNec
       }
 
+    def editOrCreateSciConfig(ed: Option[Option[Either[ScienceConfigurationModel, State[ScienceConfigurationModel, Unit]]]])(m: Option[ScienceConfigurationModel]): Option[ScienceConfigurationModel] =
+      ed.flatMap {
+        case Some(Left(s))  => s.some
+        case Some(Right(s)) => m.map(s.runS(_).value)
```

Comment:
If you choose to edit but the science configuration doesn't exist yet, then essentially nothing happens correct?

Reply:
Right, It is ignored, as we have discussed an alternative would be for it to fail in that case

Reply:
Let's go with this as I'm not sure what the "right" answer is anyway.